third_party/terraform/resources/resource_compute_instance_from_template.go

```diff
@@ -181,6 +181,14 @@ func adjustInstanceFromTemplateDisks(d *schema.ResourceData, config *Config, it
 		// boot disk was not overridden, so use the one from the instance template
 		for _, disk := range it.Properties.Disks {
 			if disk.Boot {
+				if disk.Source != "" {
+					// Instances need a URL for the disk, but instance templates only have the name
+					source, err := ParseDiskFieldValue(disk.Source, d, config)
```

Comment:
It makes perfect sense to use field values here, but that's different from what we do later in the same function. Could you adjust one to be consistent with the other?